### ISSUE TITLE
Cant turn off the Nvidia GPU

### ISSUE THREAD

**Issue by SasakiSadako** [andrebrait/prime-indicator#26]
Im Using a MSI Apache pro GE62 with an Nvidia 970m GTX.

Your Prime Indicator is truly awesome, but i would love to try the shutting off feature of your Application.

I can open the menu, but when im click on shut GPU off, nothing happends. im usualy running my KDE Neon Desktop on my Intel GPU, so i would love to turn it off.

I can push the "Eanble nvdia power management" but of course this is not what i tried to do.

Do you need any logfiles? please tell me wich files could help.

Best wishes and thank you for your great work.

Sada

**Comment by andrebrait** [comment]
How did you install it and which distro/version are you running?

**Comment by SasakiSadako** [comment]
Im using kde Neon based on Kubuntu Xenial. Its the regular Kubuntu Xenial Installation, with the KDE Neon Upgrade on it.

Im Using your version from the PPA.

Another issue ive found out is, sometimes when i want to switch the GPU with your Indicator, it tells me i have to log out and log in, but nothing happend after ive done that,

When i change the GPU from the Nvidia Panel, it always asks me about the Root Passwort, but your indicator does not.
no need for the passwort with your way?

Thank you for your great work

Sada

**Comment by andrebrait** [comment]
Threre must be something different with your system for it not to work. You're able to switch through the control panel, right?

My script uses a custom sudoers file which enables it to run as root without asking for a password.

**Comment by SasakiSadako** [comment]
Yes switch from Intel to nvidia  though the nvidia panel is no Problem.  When im using your indicator i can logout and login. Your indicator said unknown graphics Adapter but with glxinfo it said intel integrated. I have to switch from the nvidia panel for letting it work flawless and cannot shutdown the nvidia gpu

**Comment by andrebrait** [comment]
I see.

Can you post the outputs of the following commands (in order) both on Intel and on NVIDIA?

1 - cat /proc/acpi/bbswitch
2 - which prime-select
3 - which prime-supported
4 - prime-supported
5 - glxinfo | grep 'OpenGL renderer string'
6 - prime-select query

**Comment by SasakiSadako** [comment]
Here the Outputs With Intel GPU running.


```
cat /proc/acpi/bbswitch
0000:01:00.0 ON
```

```
which prime-select
/usr/bin/prime-select
```

```
which prime-supported
/usr/bin/prime-supported
```


```
prime-supported
rm: cannot remove '/var/log/prime-supported.log': Permission denied
/usr/bin/prime-supported: 35: /usr/bin/prime-supported: cannot create /var/log/prime-supported.log: Permission denied
yes

With sudo prime-supported
yes

```

```
glxinfo | grep 'OpenGL renderer string'
OpenGL renderer string: Mesa DRI Intel(R) HD Graphics 530 (Skylake GT2) 

```
```
prime-select query
intel

```

And the Output with Nvidia GPU Running

```
cat /proc/acpi/bbswitch
0000:01:00.0 ON
```


```
which prime-select
/usr/bin/prime-select
```

```
which prime-supported
/usr/bin/prime-supported
```


```
prime-supported
rm: cannot remove '/var/log/prime-supported.log': Permission denied
/usr/bin/prime-supported: 35: /usr/bin/prime-supported: cannot create /var/log/prime-supported.log: Permission denied
yes

with sudo prime-supported
yes

```

```
glxinfo | grep 'OpenGL renderer string'
OpenGL renderer string: GeForce GTX 970M/PCIe/SSE2
```

```

prime-select query
nvidia

```

Thank you for your support.

**Comment by andrebrait** [comment]
Are you using the Proprietary Drivers PPA? It seems your computer cannot
turn the NVIDIA GPU off.

Try this:

sudo tee /proc/acpi/bbswitch <<< OFF

Then post the results of

cat /proc/acpi/bbswitch

Ps.: Do this while running the Intel card.

Please provide me with your NVIDIA driver version and kernel version as
well. Post the output of the command below as well.

uname -a

Em 30 de mar de 2017 12:39 PM, "SasakiSadako" <notifications@github.com>
escreveu:

**Comment by SasakiSadako** [comment]
Here the Output and more Infos about my Driver and Kernel.

Thank you very much

```
sudo tee /proc/acpi/bbswitch <<< OFF
[sudo] password for sadako: 
OFF

```
```
cat /proc/acpi/bbswitch
0000:01:00.0 ON

```


Im using the NVIDIA Driver from the regular Ubuntu Repositoy

`The Version is 378.13
`

```
uname -a
Linux Yukie 4.8.0-36-generic #36~16.04.1-Ubuntu SMP Sun Feb 5 09:39:57 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```


I Checkout a little bit by myself and found this topic:

i think its the same issue, but with bumblebee instead of prime.
http://askubuntu.com/questions/735046/bumblebee-not-disabeling-nvidia

**Comment by andrebrait** [comment]
While running Intel, run this command and post back the output here, please (you may need to use sudo):

rmmod nvidia_drm nvidia_modeset nvidia_uvm nvidia

**Comment by SasakiSadako** [comment]
Hello

Here is my Output for sudo rmmod nvidia_drm nvidia_modeset nvidia_uvm nvidia

`rmmod: ERROR: Module nvidia_drm is not currently loaded
rmmod: ERROR: Module nvidia_modeset is not currently loaded
rmmod: ERROR: Module nvidia_uvm is not currently loaded
rmmod: ERROR: Module nvidia is not currently loaded
`

thank you for your great support.


Sada

**Comment by andrebrait** [comment]
@SasakiSadako it seems your laptop is a peculiar one... The NVIDIA's modules are not loaded, which is usually the primary cause for the card not to shut down. It seems bbswitch simply doesn't work...

Are you using the latest BIOS?
Since the main cause seems to be between your laptop/card and bbswitch, I can't really do much. Maybe the guys over the bumblebee project can help, if you open an issue there.

One last thing you can try is to update your installation to the HWE branch (the one used in Ubuntu 16.04.2) which will upgrade your kernel and drivers (both open and closed-source ones, if available):
```
sudo apt-get install --install-recommends linux-generic-hwe-16.04 xserver-xorg-hwe-16.04 
```
If this doesn't work, report your issue here:
https://github.com/Bumblebee-Project/bbswitch/issues

**Comment by SasakiSadako** [comment]
Im Using the latest bios "unlocked" and latest nvidia GPU Bios Unlocked.

The Bumblebee will not help, because bumblebee never really worked very well on my notebook.
I will checkout the HWE Branch, but anyway thank you for your great support :) and even it seems not to work with my device, keep on the great work.

**Comment by andrebrait** [comment]
I see. Are you able to disable the GPU somehow? Have you tried distros that officially support Optimus laptops, like Manjaro? If even Manjaro can't disable your GPU, I'm afraid it can't be done with Linux for some reason.

My PRIME Indicator relies on bbswitch, so it won't work unless you get bbswitch to work with your card, I'm afraid 😞

**Comment by SasakiSadako** [comment]
No problem , it's a gaming notebook and on battery with Intel I have nearly 4 hours of battery on kde neon while watching videos. I think it's a really good performance for a gaming Notebook.
No I. Not try other distros I'm using my kde neon for work and Internet and it runs stable and just windows for gaming. I found your indicator and thought maybe it would run with my device. If not, it's not just a big issue I think. 
I'm using your indicator now for the Intel / nvidia icon on the panel :) its great and enough for now 
Thank you for your support and always quick replies

**Comment by pulver22** [comment]
Hi,

I am actually having the same problem with Ubuntu 16.04.5, kernel 4.15.0.42 and nvidia driver 410.79. 
I login using the Nvidia card and then I can switch to Intel from the applet. When I relogin after the logout, the Nvidia is still on and I cannot power it down. Clicking on that option once doesn't produce any result, doing it twice freeze the pc. The same happens when I do `sudo tee /proc/acpi/bbswitch <<< OFF` . I also noticed that if I poweroff the pc while using the Intel, I have a freeze at the next login page and I must `prime-select nvidia` from the recovery mode. I guess it is not related to the applet itself but I thought it was worth to share.

**Comment by andrebrait** [comment]
They pretty much screwed up with the newer kernels and drivers.

I'll have to rewrite this indicator, soon.

**Comment by pulver22** [comment]
Good to know you are aware of the problem and willing to fix it :)
